### PR TITLE
Feature/time t nanosec [5108]

### PR DESCRIPTION
**Pull request by LuisGP**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1505/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1470/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1481/)

File: src/cpp/qos/ParameterList.cpp [hunk 1]
```diff
@@ -374,7 +374,7 @@ int32_t ParameterList::readParameterListfromCDRMsg(CDRMessage_t*msg, ParameterLi
                         }
                         DeadlineQosPolicy* p= new DeadlineQosPolicy();
                         valid &= CDRMessage::readInt32(msg,&p->period.seconds);
-                        valid &= CDRMessage::readUInt32(msg,&p->period.fraction);
+                        valid &= CDRMessage::readUInt32(msg,&p->period.nanosec);
```
Comment (richiware) [inline src/cpp/qos/ParameterList.cpp]: Duration_t at RTPS level must to be serialized using fractions.

File: src/cpp/qos/ParameterList.cpp [hunk 2]
```diff
@@ -385,7 +385,7 @@ int32_t ParameterList::readParameterListfromCDRMsg(CDRMessage_t*msg, ParameterLi
                         }
                         LatencyBudgetQosPolicy* p = new LatencyBudgetQosPolicy();
                         valid &= CDRMessage::readInt32(msg,&p->duration.seconds);
-                        valid &= CDRMessage::readUInt32(msg,&p->duration.fraction);
+                        valid &= CDRMessage::readUInt32(msg,&p->duration.nanosec);
```
Comment (richiware) [inline src/cpp/qos/ParameterList.cpp]: Duration_t at RTPS level must to be serialized using fractions.

File: src/cpp/qos/ParameterList.cpp [hunk 3]
```diff
@@ -398,7 +398,7 @@ int32_t ParameterList::readParameterListfromCDRMsg(CDRMessage_t*msg, ParameterLi
                         valid&=CDRMessage::readOctet(msg,(octet*)&p->kind);
                         msg->pos+=3;
                         valid &= CDRMessage::readInt32(msg,&p->lease_duration.seconds);
-                        valid &= CDRMessage::readUInt32(msg,&p->lease_duration.fraction);
+                        valid &= CDRMessage::readUInt32(msg,&p->lease_duration.nanosec);
```
Comment (richiware) [inline src/cpp/qos/ParameterList.cpp]: Duration_t at RTPS level must to be serialized using fractions.

File: src/cpp/qos/ParameterList.cpp [hunk 4]
```diff
@@ -422,7 +422,7 @@ int32_t ParameterList::readParameterListfromCDRMsg(CDRMessage_t*msg, ParameterLi
                         valid&=CDRMessage::readOctet(msg,(octet*)&p->kind);
                         msg->pos+=3;
                         valid &= CDRMessage::readInt32(msg,&p->max_blocking_time.seconds);
-                        valid &= CDRMessage::readUInt32(msg,&p->max_blocking_time.fraction);
+                        valid &= CDRMessage::readUInt32(msg,&p->max_blocking_time.nanosec);
```
Comment (richiware) [inline src/cpp/qos/ParameterList.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/ParameterList.cpp [hunk 5]
```diff
@@ -480,7 +480,7 @@ int32_t ParameterList::readParameterListfromCDRMsg(CDRMessage_t*msg, ParameterLi
                         }
                         TimeBasedFilterQosPolicy* p = new TimeBasedFilterQosPolicy();
                         valid &= CDRMessage::readInt32(msg,&p->minimum_separation.seconds);
-                        valid &= CDRMessage::readUInt32(msg,&p->minimum_separation.fraction);
+                        valid &= CDRMessage::readUInt32(msg,&p->minimum_separation.nanosec);
```
Comment (richiware) [inline src/cpp/qos/ParameterList.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/ParameterList.cpp [hunk 6]
```diff
@@ -571,7 +571,7 @@ int32_t ParameterList::readParameterListfromCDRMsg(CDRMessage_t*msg, ParameterLi
                         }
                         DurabilityServiceQosPolicy * p = new DurabilityServiceQosPolicy();
                         valid &= CDRMessage::readInt32(msg,&p->service_cleanup_delay.seconds);
-                        valid &= CDRMessage::readUInt32(msg,&p->service_cleanup_delay.fraction);
+                        valid &= CDRMessage::readUInt32(msg,&p->service_cleanup_delay.nanosec);
```
Comment (richiware) [inline src/cpp/qos/ParameterList.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/ParameterList.cpp [hunk 7]
```diff
@@ -587,7 +587,7 @@ int32_t ParameterList::readParameterListfromCDRMsg(CDRMessage_t*msg, ParameterLi
                         }
                         LifespanQosPolicy * p = new LifespanQosPolicy();
                         valid &= CDRMessage::readInt32(msg,&p->duration.seconds);
-                        valid &= CDRMessage::readUInt32(msg,&p->duration.fraction);
+                        valid &= CDRMessage::readUInt32(msg,&p->duration.nanosec);
```
Comment (richiware) [inline src/cpp/qos/ParameterList.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/QosPolicies.cpp [hunk 8]
```diff
@@ -42,7 +42,7 @@ bool DeadlineQosPolicy::addToCDRMessage(CDRMessage_t* msg)
     bool valid = CDRMessage::addUInt16(msg, this->Pid);
     valid &= CDRMessage::addUInt16(msg, this->length);//this->length);
     valid &= CDRMessage::addInt32(msg,period.seconds);
-    valid &= CDRMessage::addUInt32(msg,period.fraction);
+    valid &= CDRMessage::addUInt32(msg,period.nanosec);
```
Comment (richiware) [inline src/cpp/qos/QosPolicies.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/QosPolicies.cpp [hunk 9]
```diff
@@ -51,7 +51,7 @@ bool LatencyBudgetQosPolicy::addToCDRMessage(CDRMessage_t* msg) {
     bool valid = CDRMessage::addUInt16(msg, this->Pid);
     valid &= CDRMessage::addUInt16(msg, this->length);//this->length);
     valid &= CDRMessage::addInt32(msg,duration.seconds);
-    valid &= CDRMessage::addUInt32(msg,duration.fraction);
+    valid &= CDRMessage::addUInt32(msg,duration.nanosec);
```
Comment (richiware) [inline src/cpp/qos/QosPolicies.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/QosPolicies.cpp [hunk 10]
```diff
@@ -63,7 +63,7 @@ bool LivelinessQosPolicy::addToCDRMessage(CDRMessage_t* msg) {
     valid &= CDRMessage::addOctet(msg,0);
     valid &= CDRMessage::addOctet(msg,0);
     valid &= CDRMessage::addInt32(msg,lease_duration.seconds);
-    valid &= CDRMessage::addUInt32(msg,lease_duration.fraction);
+    valid &= CDRMessage::addUInt32(msg,lease_duration.nanosec);
```
Comment (richiware) [inline src/cpp/qos/QosPolicies.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/QosPolicies.cpp [hunk 11]
```diff
@@ -85,7 +85,7 @@ bool ReliabilityQosPolicy::addToCDRMessage(CDRMessage_t* msg) {
     valid &= CDRMessage::addOctet(msg,0);
     valid &= CDRMessage::addOctet(msg,0);
     valid &= CDRMessage::addInt32(msg,max_blocking_time.seconds);
-    valid &= CDRMessage::addUInt32(msg,max_blocking_time.fraction);
+    valid &= CDRMessage::addUInt32(msg,max_blocking_time.nanosec);
```
Comment (richiware) [inline src/cpp/qos/QosPolicies.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/QosPolicies.cpp [hunk 12]
```diff
@@ -103,7 +103,7 @@ bool TimeBasedFilterQosPolicy::addToCDRMessage(CDRMessage_t* msg) {
     bool valid = CDRMessage::addUInt16(msg, this->Pid);
     valid &= CDRMessage::addUInt16(msg, this->length);//this->length);
     valid &= CDRMessage::addInt32(msg,minimum_separation.seconds);
-    valid &= CDRMessage::addUInt32(msg,minimum_separation.fraction);
+    valid &= CDRMessage::addUInt32(msg,minimum_separation.nanosec);
```
Comment (richiware) [inline src/cpp/qos/QosPolicies.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/QosPolicies.cpp [hunk 13]
```diff
@@ -190,7 +190,7 @@ bool DurabilityServiceQosPolicy::addToCDRMessage(CDRMessage_t* msg) {
     bool valid = CDRMessage::addUInt16(msg, this->Pid);
     valid &= CDRMessage::addUInt16(msg, this->length);//this->length);
     valid &= CDRMessage::addInt32(msg,service_cleanup_delay.seconds);
-    valid &= CDRMessage::addUInt32(msg,service_cleanup_delay.fraction);
+    valid &= CDRMessage::addUInt32(msg,service_cleanup_delay.nanosec);
```
Comment (richiware) [inline src/cpp/qos/QosPolicies.cpp]: Duration_t at RTPS level must be serialized using fractions.

File: src/cpp/qos/QosPolicies.cpp [hunk 14]
```diff
@@ -206,7 +206,7 @@ bool LifespanQosPolicy::addToCDRMessage(CDRMessage_t* msg) {
     bool valid = CDRMessage::addUInt16(msg, this->Pid);
     valid &= CDRMessage::addUInt16(msg, this->length);//this->length);
     valid &= CDRMessage::addInt32(msg,duration.seconds);
-    valid &= CDRMessage::addUInt32(msg,duration.fraction);
+    valid &= CDRMessage::addUInt32(msg,duration.nanosec);
```
Comment (richiware) [inline src/cpp/qos/QosPolicies.cpp]: Duration_t at RTPS level must be serialized using fractions.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1517/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1481/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1492/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1533/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1496/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1507/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1534/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1497/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1508/)

**Comment by raquelalvarezbanos** [comment]
@LuisGP Some conflicts arose probably due to Deadline QoS branch. Other than this, requested changes were resolved, so this is good to merge.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1543/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1506/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1517/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1550/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1513/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1524/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1553/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1516/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1527/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1564/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1528/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1539/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1594/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1558/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1569/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1597/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1561/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1572/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1609/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1573/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1584/)

**Comment by richiware** [comment]
Failed tests not related with this PR.

**Pull request merged**